include/requests.hpp

```diff
@@ -28,7 +32,18 @@ struct Request {
 using RequestPtr = std::shared_ptr<Request>;
 
 struct ParameterValueRequest : Request {
-    explicit ParameterValueRequest(const std::vector<std::string> &parameterNames_) : Request(Type::QUERY_PARAMETER_VALUES), parameterNames(parameterNames_) {};
+    explicit ParameterValueRequest(
+        const std::vector<std::string> &parameterNames_, 
+        const std::vector<rcl_interfaces::msg::ParameterDescriptor> &parameterDescriptors_) 
+            : Request(Type::QUERY_PARAMETER_VALUES), parameterNames(parameterNames_),
+              parameterDescriptors(parameterDescriptors_) {};
+
+    std::vector<std::string> parameterNames;
+    std::vector<rcl_interfaces::msg::ParameterDescriptor> parameterDescriptors;
```

Comment:
why does the value request need the descriptors?

Reply:
This is for convenience so that the descriptors can be passed on when the values are received:

[service_wrapper.cpp:L203](https://github.com/teamspatzenhirn/rig_reconfigure/pull/42/files#diff-237bafec7ef58175842b31b7c6716d0e1364dcce3e4bf36df986c82abf0f6e76R203)

and then here

[service_wrapper.cpp:L300](https://github.com/teamspatzenhirn/rig_reconfigure/pull/42/files#diff-237bafec7ef58175842b31b7c6716d0e1364dcce3e4bf36df986c82abf0f6e76R300)

Open to other ways of doing this.

Reply:
no problem if it makes sense, just seemed nonintuitive when looking at it